ui/src/app/utils/workspace-utils.tsx

```diff
@@ -1,16 +1,31 @@
-import { FeaturedWorkspaceCategory, Workspace } from 'generated/fetch';
+import {
+  BillingStatus,
+  FeaturedWorkspaceCategory,
+  Workspace,
+} from 'generated/fetch';
 
 import { serverConfigStore } from 'app/utils/stores';
 
-export const isUsingFreeTierBillingAccount = (
-  workspace: Workspace
-): boolean => {
+export const isUsingInitialCredits = (workspace: Workspace): boolean => {
   return (
     workspace.billingAccountName ===
     'billingAccounts/' + serverConfigStore.get().config.freeTierBillingAccountId
   );
 };
 
+export const isValidBilling = (workspace: Workspace): boolean => {
+  const isInitialCredits = isUsingInitialCredits(workspace);
+  const enableInitialCreditsExpiration =
+    serverConfigStore.get().config.enableInitialCreditsExpiration;
+  return enableInitialCreditsExpiration
+    ? (isInitialCredits &&
+        !workspace.initialCredits.exhausted &&
+        (!workspace.initialCredits.expired ||
+          workspace.initialCredits.expirationBypassed)) ||
+        !isInitialCredits
```

Comment:
IMO the logic would be a little easier to read if `!isInitialCredits` was the first clause, like:
```
!isInitialCredits ||
  (isInitialCredits &&
        !workspace.initialCredits.exhausted &&
        (!workspace.initialCredits.expired ||
          workspace.initialCredits.expirationBypassed))
```